### PR TITLE
[Semantic Segmentation] Add Interface for Blueoil

### PR DESCRIPTION
**Pull request by iizukak**
## Motivation and Context

This PR is part of #88 

## Description

- Add Semantic Segmentation Interface to Blueoil
- Add  test using`blueoil_test.sh`

## How has this been tested?

This PR contain test with `blueoil_test.sh`, and I manually tested using #174 tutorial documentation.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: blueoil/blueoil_init.py [hunk 1]
```diff
@@ -76,6 +70,10 @@
         "max_size": 480,
         "divider": 32,
     },
+    "LmSegnetV1Quantize": {
+        "max_size": 512,
+        "divider": 1,
```
Comment (ruimashita) [inline blueoil/blueoil_init.py]: The `divider` is for restriction of input image size, the image need to be able to divide `divider` value. `LmSegnetV 1 Quantize` network is used 3 time `s2d`, then `divider` is 2^3 = `8`

File: blueoil/templates/lmnet/semantic_segmentation.tpl.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+PRETRAIN_FILE = ""
+
+# for debug
+# BATCH_SIZE = 2
+# SUMMARISE_STEPS = 1
+# IS_DEBUG = True
+
+PRE_PROCESSOR = Sequence([
+    Resize(size=IMAGE_SIZE),
+    DivideBy255(),
```
Comment (ruimashita) [inline blueoil/templates/lmnet/semantic_segmentation.tpl.py]: The `PRE_PROCESSOR` need to fit other templates.
https://github.com/blue-oil/blueoil/blob/master/blueoil/templates/lmnet/object_detection.tpl.py#L70

**Comment by iizukak** [comment]
@ruimashita Thanks! I'll fix your commented issues.
After some PR #177 #188 , This PR is not working now.
And I re added `[WIP]`. I will remove `[WIP]` after I complete this PR.

**Comment by ruimashita** [comment]
@iizukak Yes sir 👍

File: blueoil/templates/lmnet/semantic_segmentation.tpl.py (hunk 2 above)
Comment (iizukak) [reply blueoil/templates/lmnet/semantic_segmentation.tpl.py]: Fixed.

File: blueoil/blueoil_init.py (hunk 1 above)
Comment (iizukak) [reply blueoil/blueoil_init.py]: Fixed.

**Comment by iizukak** [comment]
@ruimashita 
I fixed your commented issues and some errors around `blueoil_test.sh`.
I also manually tested `init`, `train` and `convert` can run.
Please review again 🙇

**Comment by iizukak** [comment]
@ruimashita Please review again 🙇

File: blueoil/templates/lmnet/semantic_segmentation.tpl.py (hunk 2 above)
Comment (ruimashita) [reply blueoil/templates/lmnet/semantic_segmentation.tpl.py]: 👍

File: blueoil/templates/lmnet/semantic_segmentation.tpl.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+NETWORK.BATCH_SIZE = BATCH_SIZE
+NETWORK.DATA_FORMAT = DATA_FORMAT
+NETWORK.ACTIVATION_QUANTIZER = linear_mid_tread_half_quantizer
+NETWORK.ACTIVATION_QUANTIZER_KWARGS = {
+    'bit': 2,
+    'max_value': 2
+}
+NETWORK.WEIGHT_QUANTIZER = binary_mean_scaling_quantizer
+NETWORK.WEIGHT_QUANTIZER_KWARGS = {}
+NETWORK.LABEL_COLORS = get_color_map(len({{classes}}))
```
Comment (ruimashita) [inline blueoil/templates/lmnet/semantic_segmentation.tpl.py]: The `NETWORK.LABEL_COLORS` parameter can be hidden if you change source code on segmentation `Base` class in `lmnet/networks/segmentation/base.py` like below

```
    def __init__(
            self,
            *args,
            label_colors=None,
            **kwargs
    ):
        super().__init__(
            *args,
            **kwargs,
        )
        if label_colors is None:
            self.label_colors = get_color_map(self.num_classes)
        else:
            self.label_colors = label_colors
```

Less parameter is good for maintenance I think.

Comment (iizukak) [reply blueoil/templates/lmnet/semantic_segmentation.tpl.py]: @ruimashita Thanks. Fixed.

**Comment by iizukak** [comment]
@ruimashita 
I removed your noticed lines in this commit.
https://github.com/blue-oil/blueoil/pull/180/commits/113d026000d20c03033598789bc71d2b54695f66

**Pull request merged**